src/robot/dial.ts

```diff
@@ -185,3 +193,44 @@ export const createRobotClient = async (
 
   return client;
 };
+
+export const createViamTransportFactory = async (
+  serviceHost: string,
+  dialOpts: DialOptions
+): Promise<grpc.TransportFactory> => {
+  if (!dialOpts.credentials) {
+    throw new Error(`credential cannot be none`);
+  } else if (dialOpts.credentials.type === 'robot-secret') {
+    throw new Error(`credential type cannot be 'robot secret'`);
+  } else if (!dialOpts.authEntity) {
+    throw new Error(`auth entity cannot be None`);
+  }
+
+  const entity = dialOpts.authEntity;
+  const creds = new Credentials();
+  creds.setType(dialOpts.credentials.type);
+  creds.setPayload(dialOpts.credentials.payload);
+
+  const req = new AuthenticateRequest();
+  req.setEntity(entity);
+  req.setCredentials(creds);
+
+  const transportFactory = await nodeDialDirect(serviceHost);
+  const authClient = new AuthServiceClient(serviceHost, {
+    transport: transportFactory,
+  });
+
+  const accessToken = await new Promise<string>((resolve, reject) => {
+    authClient.authenticate(req, new grpc.Metadata(), (err, response) => {
+      if (err) {
+        return reject(err);
+      }
+      const token = response?.getAccessToken().toString() ?? '';
```

Comment:
Would an empty access token be considered an authentication error?

Reply:
According to @cheukt, it will throw an authentication error elsewhere, so there isn't a pressing need for there to be a check here.